test/integration/user-management-api/base_controller_test.rb

```diff
@@ -29,6 +29,74 @@ def test_unknown_format
     end
   end
 
+  class RepresentedPaginationMetadataTest < ActionDispatch::IntegrationTest
+    include RepresentedApiRouting
+    disable_transactional_fixtures!
+
+    def setup
+      provider = FactoryBot.create(:provider_account)
+      login! provider
+    end
+
+    test 'JSON/XML pagination metadata for representer class' do
+      with_api_routes do
+        get '/api/klas', params: { format: :json, words: %w[hello world example] }
+        response_hash = JSON.parse(response.body)
+        assert_nil response_hash['metadata']
+
+        get '/api/klas', params: { format: :json, words: %w[hello world example], per_page: 1, page: 2 }
+        response_hash = JSON.parse(response.body)
+        assert_equal 1, response_hash.dig('metadata', 'per_page')
+        assert_equal 3, response_hash.dig('metadata', 'total_entries')
+        assert_equal 3, response_hash.dig('metadata', 'total_pages')
+        assert_equal 2, response_hash.dig('metadata', 'current_page')
+
+        get '/api/klas', params: { format: :xml, words: %w[hello world example] }
+        response_hash = Hash.from_xml(response.body)
+        assert_nil response_hash.dig('klass', 'per_page')
+        assert_nil response_hash.dig('klass', 'total_entries')
+        assert_nil response_hash.dig('klass', 'total_pages')
+        assert_nil response_hash.dig('klass', 'current_page')
+
+        get '/api/klas', params: { format: :xml, words: %w[hello world example], per_page: 1, page: 2 }
+        response_hash = Hash.from_xml(response.body)
+        assert_equal '1', response_hash.dig('klass', 'per_page')
+        assert_equal '3', response_hash.dig('klass', 'total_entries')
+        assert_equal '3', response_hash.dig('klass', 'total_pages')
+        assert_equal '2', response_hash.dig('klass', 'current_page')
+      end
+    end
+
+    test 'JSON/XML pagination metadata for representer module' do
+      with_api_routes do
+        get '/api/mods', params: { format: :json, words: %w[hello world example] }
+        response_hash = JSON.parse(response.body)
+        assert_nil response_hash['metadata']
+
+        get '/api/mods', params: { format: :json, words: %w[hello world example], per_page: 1, page: 2 }
+        response_hash = JSON.parse(response.body)
+        assert_equal 1, response_hash.dig('metadata', 'per_page')
+        assert_equal 3, response_hash.dig('metadata', 'total_entries')
+        assert_equal 3, response_hash.dig('metadata', 'total_pages')
+        assert_equal 2, response_hash.dig('metadata', 'current_page')
+
+        get '/api/mods', params: { format: :xml, words: %w[hello world example] }
+        response_hash = Hash.from_xml(response.body)
+        assert_nil response_hash.dig('mods', 'per_page')
```

Comment:
Shouldn't this be in `metadata`?

Reply:
No because in the xml is in the attributes. Look at the example in the description of the PR:
![image](https://user-images.githubusercontent.com/11318903/101054275-01af7600-3589-11eb-919f-e3ee1573821e.png)

Reply:
so first the name of the collection (which is the name of the model in plural) and then the attribute itself